py_feels/binary/training/binary_classifier_training.py

```diff
@@ -2,102 +2,198 @@
 """
 Created on Thu Dec  7 15:45:21 2017
 
-Training script for binary text classifier
+Training script for binary text classifier. neutral class added
+for accuracy
+
+positive text is given a value of +1
+neutral  text is given a value of  0
+negative text is given a value of -1
 
 @author: rcehemann
 """
 ###############################################################################
 
 # logistical stuff
 import pandas as pd                                     # pandas allows easy manipulation of the data
-from sklearn.feature_extraction.text import TfidfTransformer, CountVectorizer # preprocessors
+import numpy as np                                      # 
+from sklearn.feature_extraction.text import CountVectorizer # preprocessor
 from sklearn.feature_selection import chi2, SelectKBest # used to filter useless data
 from sklearn.pipeline import Pipeline                   # pre-processing pipeline
 from sklearn.model_selection import cross_val_score     # for testing accuracy
 from sklearn.externals.joblib import dump               # fitted model will be pickled for later use
 
-# classification algorithms
-from sklearn.svm      import SVC                        # support-vector machine classifier
-from sklearn.ensemble import RandomForestClassifier, GradientBoostingClassifier # tree-based classifiers
-from sklearn.neural_network import MLPClassifier        # multilayer perceptron classifier (dense NN)
-from sklearn.naive_bayes import MultinomialNB           # naive bayes classifier w/ multinomial distribution
+# classification algorithm
+from sklearn.naive_bayes import MultinomialNB # naive bayes classifiers
 
 ###############################################################################
-
-
+#%%
+""" this section assembles training data """
 # assemble training data from various files into pandas DataFrame objects.
-# format of the files is a sentence followed by a sentiment
-# 1 == positive sentiment and 0 == negative sentiment
+# format of the dataframe is a sentence followed by a sentiment
+# 1 == positive sentiment and 0 == neutral and -1 == negative
+
+#<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+# First UCI data from amazon/yelp/imdb
 amazon = pd.read_csv(
         'UCI_training_data/amazon_cells_labelled.txt', 
         sep="\t", 
         names=['sentence', 'sentiment'], 
         dtype={'sentence':str, 'sentiment':int}
 )
+# convert from 1/0 scheme to 1/0/-1 scheme
+amazon.sentiment = 2 * amazon.sentiment - 1
```

Comment:
:suspect: 
Scheme conversion functions? If you ever wanted to change the scheme again, you'd have three places to update.